parent_swap/swap.py

```diff
@@ -1,22 +1,43 @@
+from django.db import models
 from django.conf import settings
+from django.views.generic.detail import DetailView
 
+from rest_framework import serializers
 
-def get_class_object(path):
-    """
-    Returns the class instance of the app
-    """
-    module_name, sp, cls_name = path.rpartition('.')
-    mod = __import__(module_name, fromlist=[cls_name])
-    return getattr(mod, cls_name)
 
+__all__ = ['BaseModel', 'BaseMapping', 'BaseSerializer', 'BaseView']
 
-def get_default_base_class():
-    """
-    Returns the class reference in settings.DEFAULT_BASE_CLASS
-    """
-    base_cls_path = getattr(settings, 'DEFAULT_BASE_CLASS', 'django.db.models.Model')
-    return get_class_object(base_cls_path)
 
+def import_class(cls_path, default=None):
+    try:
+        mod_name, sp, cls_name = cls_path.rpartition('.')
+        mod = __import__(mod_name, fromlist=[''])
+        cls_obj = getattr(mod, cls_name, default)
+        return cls_obj
+    except:
+        return default
 
-"""Our sweet sweet BaseClass."""
-BaseClass = get_default_base_class()
+
+def get_base_model():
+    default_model_name = getattr(settings, 'DEFAULT_BASE_MODEL', None)
+    return import_class(default_model_name, models.Model)
+
+
+def get_base_mapping(base_cls):
+    return getattr(base_cls, 'Mapping', object)
+
+
+def get_base_serializer():
+    default_serializer_name = getattr(settings, 'DEFAULT_SERIALIZER', None)
```

Comment:
Seems like all of these settings should all have _BASE_ in their name to be consistent.